### ISSUE TITLE
Make permission system more fine-grained

### ISSUE THREAD

**Issue by RJanssens** [goharbor/harbor#12062]
In our Harbor instance, we'd like to allow users to create and manage their own projects. However, deployment security should remain enabled in all cases. 

Is it possible to allow us to either define custom roles (like an admin 'light' who cannot change some configuration settings) or adapt existing roles (so system administrators can define which rights the default roles have)?

**Comment by ninjaslothx** [comment]
It would also be nice to have a RO administrator role as mentioned in https://github.com/goharbor/harbor/issues/15711.
However being able to tune these roles would be ideal. 
For example often a security team might need the ability to run scanning jobs on any project/repo + manage vulnerability exceptions.

**Comment by gwiersma** [comment]
It would also be nice if users could create there own replication rules and endpoint registries.

**Comment by slushysnowman** [comment]
Yeah we're running into some areas where it would be really nice if we could create custom rules for our users. For example, potentially we want some users to create robot accounts, but not allow them to set a project to 'public' - at the moment as far as I can see, this is not possible in the current permissions model.
